src/factory/src/api.rs

```diff
@@ -121,6 +121,30 @@ impl TokenFactoryCanister {
 
         Ok(principal)
     }
+
+    /// Delete a token.
+    /// The token must be owned by the caller.
+    /// The token will be deleted from the factory and the removed from the canister registry.
+    #[update]
+    async fn delete_token(&self, name: String) -> Result<(), TokenFactoryError> {
+        //    Check controller access
+        let state_ref = &mut *self.state.borrow_mut();
+        state_ref.check_controller_access()?;
+
+        let token = self.get_token(name.clone()).await;
+
+        if token.is_none() {
+            return Err(TokenFactoryError::FactoryError(FactoryError::NotFound));
+        }
+
+        state_ref
+            .factory()
+            .drop(token.expect("token exists, checked above"))
+            .await?;
```

Comment:
Rather than checking that the token `is_some` and then unwrapping (`expect`), we can make sure that the token is either a token or we do an early return with an error:

```suggestion
        let token = match self.get_token(name.clone()).await {
            Some(token) => token,
            None => return Err(TokenFactoryError::FactoryError(FactoryError::NotFound)),
        };

        state_ref
            .factory()
            .drop(token)
            .await?;
```

Reply:
Yeah this makes better sense